core/repository/api/src/main/java/org/eclipse/rdf4j/repository/util/RDFLoader.java

```diff
@@ -87,7 +93,10 @@ public void load(File file, String baseURI, RDFFormat dataFormat, RDFHandler rdf
 	}
 
 	/**
-	 * Parses the RDF data that can be found at the specified URL to the RDFHandler.
+	 * Parses the RDF data that can be found at the specified URL to the RDFHandler. This method uses the class
+	 * {@link URL} to resolve the provided <tt>url</tt>. While this class doesn't follow redirects upgrading the
+	 * protocol from HTTP to HTTPS, this method implements its own strategy to follow redirects that also supports this
+	 * kind of insecure request upgrades.
```

Comment:
I'd reformulate slightly, to something like this:

> This method honors HttpUrlConnection.getFollowRedirects to determine if redirects are followed and if set to true will also follow redirects from http to https. The maximum number of redirects can be controlled using system property `http.maxRedirects`.

Reply:
I can edit the Javadoc according to your suggestions. Concerning the unit tests:
* where should I add the unit tests (i.e. module, and fully qualified class name)?
* I had in mind to use something like [Mock Server](https://www.mock-server.com/). Can I use it, or do you already use something else? In the latter case, can you point me to some example test

Reply:
Great, thanks! 

Unit tests go in the same module, in src/test/java, the only exception being tests where other parts of the framework are needed that would introduce a circular dependency: in those cases, the test is more like an integration tests, and goes into the compliance modules. In this case, I think you should be fine with just a unit test in the repository-api module itself. Name it `org.eclipse.rdf4j.repository.util.RDFLoaderTest`. 

As for mocking, I actually had in mind to just use Mockito to mock out the connection behavior at the client side, but if you prefer to mock the server-side instead, we use [WireMock](https://www.baeldung.com/introduction-to-wiremock) in several places for this purpose. Have a look at [SPARQLProtocolSessionTest](https://github.com/eclipse/rdf4j/blob/main/core/http/client/src/test/java/org/eclipse/rdf4j/http/client/SPARQLProtocolSessionTest.java) or [RemoteRepositoryManagerTest](https://github.com/eclipse/rdf4j/blob/main/core/repository/manager/src/test/java/org/eclipse/rdf4j/repository/manager/RemoteRepositoryManagerTest.java) for some examples.